lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -307,6 +315,98 @@ public void OnCompleted()
         {
         }
 
+        /// <summary>
+        /// Generates map task configuration given the active context. 
+        /// Merge configurations of all the inputs to the MapTaskHost.
+        /// </summary>
+        /// <param name="activeContext">Active context to which task needs to be submitted</param>
+        /// <param name="taskId">Task Id</param>
+        /// <returns>Map task configuration</returns>
+        private IConfiguration GetMapTaskConfiguration(IActiveContext activeContext, string taskId)
+        {
+            IConfiguration mapSpecificConfig;
+
+            if (!_perMapperConfiguration.TryPop(out mapSpecificConfig))
+            {
+                Exceptions.Throw(
+                    new IMRUBugException(string.Format("No per map configuration exist for the active context {0}",
+                        activeContext.Id)),
+                    Logger);
+            }
+
+            return TangFactory.GetTang()
+                .NewConfigurationBuilder(TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, taskId)
+                    .Set(TaskConfiguration.Task, GenericType<MapTaskHost<TMapInput, TMapOutput>>.Class)
+                    .Build(),
+                    _configurationManager.MapFunctionConfiguration,
+                    mapSpecificConfig)
+                .BindNamedParameter(typeof(InvokeGC), _invokeGC.ToString())
+                .Build();
+        }
+
+        /// <summary>
+        /// Generates the update task configuration.
+        /// Merge configurations of all the inputs to the UpdateTaskHost.
+        /// </summary>
+        /// <returns>Update task configuration</returns>
+        private IConfiguration GetUpdateTaskConfiguration()
+        {
+            var partialTaskConf =
+                TangFactory.GetTang()
+                    .NewConfigurationBuilder(TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier,
+                            IMRUConstants.UpdateTaskName)
+                        .Set(TaskConfiguration.Task,
+                            GenericType<UpdateTaskHost<TMapInput, TMapOutput, TResult>>.Class)
+                        .Build(),
+                        _configurationManager.UpdateFunctionConfiguration,
+                        _configurationManager.ResultHandlerConfiguration)
+                    .BindNamedParameter(typeof(InvokeGC), _invokeGC.ToString())
+                    .Build();
+
+            try
+            {
+                TangFactory.GetTang()
+                    .NewInjector(partialTaskConf, _configurationManager.UpdateFunctionCodecsConfiguration)
+                    .GetInstance<IIMRUResultHandler<TResult>>();
```

Comment:
You might want to add some  comments here otherwise every time, people look at this code and will ask this question.